dotcms-integration/src/test/java/com/dotmarketing/portlets/contentlet/business/HostAPITest.java

```diff
@@ -1395,4 +1394,37 @@ public void findAllFromDB() throws DotDataException, DotSecurityException {
                 siteListWithSystemHost.size() - siteList.size());
     }
 
+    /**
+     * <ul>
+     *     <li><b>Method to test: </b>{@link HostAPI#findByIdOrKey(String, User, boolean)}</li>
+     *     <li><b>Given Scenario: </b>Find a Site using either its Identifier or its Key.</li>
+     *     <li><b>Expected Result: </b>The Site API must be able to find the specified Site using
+     *     either the Identifier or the Site Key.</li>
+     * </ul>
+     */
+    @Test
+    public void findSiteByIdOrKey() throws DotDataException, DotSecurityException {
+        // ╔══════════════════╗
+        // ║  Initialization  ║
+        // ╚══════════════════╝
+        final HostAPI siteAPI = APILocator.getHostAPI();
+        final User systemUser = APILocator.systemUser();
+        final String defaultSiteId = "48190c8c-42c4-46af-8d1a-0cd5db894797";
```

Comment:
Let's create a new Site using the data-gen to avoid depending upon existing data.

Reply:
Done!